packages/website/pages/[...slug].tsx

```diff
@@ -58,76 +53,142 @@ export default function ComponentPage({
       </PropsContextProvider>
     </>
   );
+};
+
+// TODO: mrege Heading and getToC to getTableOfContents from 'mdx-utils'
+interface Heading {
+  nodeType: string;
+  content: Array<{ value: string }>;
 }
 
-export async function getStaticProps(props: { params: { slug: string[] } }) {
-  const result = await getMdxSourceBySlug(props.params.slug);
+function getToC(content) {
+  let tableOfContents: HeadingType[] = [];
+  const headings: Heading[] = content.filter((node) =>
+    node.nodeType.includes('heading'),
+  );
 
-  if (!result) {
-    throw new Error(
-      'Could not read file by slug: ' + props.params.slug.join('/'),
-    );
+  if (headings.length) {
+    tableOfContents = headings.map((heading) => {
+      const headingType = heading.nodeType === 'heading-2' ? 'h2' : 'h3';
+      const headingText = heading.content[0].value;
+      const headingLink = slugger.slug(headingText, false);
+      return {
+        text: headingText,
+        id: headingLink,
+        level: headingType,
+      };
+    });
   }
 
-  const {
-    frontMatter: { content, data },
-  } = result;
-
-  let toc = {};
-
-  let shortIntroText = '';
-  let mainContentText = content;
-
-  // It will match every text that comes before the first "## Import"
-  const shortIntroRegex = new RegExp(/([^#]+)/);
-  const matches = content.match(shortIntroRegex);
-
-  if (matches !== null) {
-    shortIntroText = matches[0];
-    mainContentText = content.replace(matches[0], '');
-  }
+  return tableOfContents;
+}
 
-  const shortIntro = await serialize(shortIntroText);
-  const mainContent = await serialize(mainContentText, {
-    // Optionally pass remark/rehype plugins
-    mdxOptions: {
-      remarkPlugins: [remarkCodeTitles, remarkCodeImport],
-      rehypePlugins: [
-        rehypeSlug,
-        [
-          rehypeToc,
-          {
-            nav: false,
-            headings: ['h1', 'h2', 'h3'],
-            customizeTOC: (t) => {
-              toc = transformToc(t) as any;
-              return false;
+// [WIP]: I started adding stronger types to this function but I could not finish
+// that's why I renamed some stuff here, I was trying to follow this:
+// https://nextjs.org/docs/basic-features/typescript#static-generation-and-server-side-rendering
+export const getStaticProps = async (context: {
```

Comment:
I would separate some parts into functions because right now it's a bit hard to read and understand. For example the part with shortIntroText can be isolated